### PR TITLE
[Oy2-19462] Dropdown Component

### PR DESCRIPTION
**Pull request by jphechter**
## Purpose

Add the dropdown component.

## Type of Request

feature

## Approach

Implements the basic dropdown functionality found in the USWDS library.

## Assorted Notes/Considerations

Testing around this component, while showing 100%, is not comprehensive. There is a single snapshot test that covers basic rendering, but does not cover true functionality of the component. We're unable to get Jest to render the component with the USWDS scripts, so we cannot actually test any of the user interactions. Once we figure this out, tests and other functionality could be written for this.

#### Pull Request Creator Checklist

- [X] It is documented what type of request this is.
- [X] There is a detailed description for the purpose of this PR.
- [X] This PR is in line with the intention of this Project

### CODE REVIEW

File: src/assets/theme/_uswds-theme-custom-styles.scss [hunk 1]
```diff
@@ -21,6 +21,7 @@ i.e.
 */
 @forward "../../components/Button/Button.scss";
 @forward "../../components/Checkbox/Checkbox.scss";
+@forward "../../components/Dropdown/Dropdown.scss";
```
Comment (jphechter) [inline src/assets/theme/_uswds-theme-custom-styles.scss:24]: This file doesn't actually exist. Will remove this line.

File: tsconfig.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
     ] /* Specify type package names to be included without being referenced in a source file. */,
     // "allowUmdGlobalAccess": true,                     /* Allow accessing UMD globals from modules. */
     // "moduleSuffixes": [],                             /* List of file name suffixes to search when resolving a module. */
-    // "resolveJsonModule": true,                        /* Enable importing .json files. */
+    "resolveJsonModule": true,                           /* Enable importing .json files. */
```
Comment (jphechter) [inline tsconfig.json:44]: This allows Jest to import the data object in `data.json`.

File: src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            </li>
+          ))}
+        </ul>
+        <div className="usa-combo-box__status usa-sr-only" role="status"></div>
+        <span className="usa-sr-only" id={`${id}--assistiveHint`}>
+          When autocomplete results are available use up and down arrows to
+          review and enter to select. Touch device users, explore by touch or
+          with swipe gestures.
+        </span>
+      </div>
```
Comment (jdinh8124) [inline src/components/Dropdown/Dropdown.tsx:131]: Was this a uswds recommendation?

**Review by jdinh8124 — APPROVED** [review]
Looks good to me

File: src/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (jphechter) [reply src/components/Dropdown/Dropdown.tsx:131]: This is part of the USWDS component.

**Pull request merged**